### DISCUSSION TITLE
SqlClient EF Code Application Intent Setting

### DISCUSSION THREAD

**Discussion by andrewluck-2degrees** [dotnet/SqlClient#2307] [Ideas]
For performance reasons in production could it be of benefit to the wider audience to have a built in way to set the application intent to use the secondary read-only nodes for select queries.
There is always the option to add a DbCommandInterceptor but this should only be applied when you not running within a transaction.
I think it would be beneficial to have a native implementation in SqlClient that can benefit many people.

**Comment by David-Engel** [comment]
I'm not sure how this would work in practice. A different application intent can require a connection to a different server. It sounds like you are basically asking for the library to dynamically run your query from a pool of connections based on the query requirement. That is very different from the current design where DbConnections are parents of DbCommands in ADO.NET.
